lib/galaxy/model/__init__.py

```diff
@@ -8921,12 +8921,13 @@ def to_dict(self, view="collection", value_mapper=None, step_details=False, lega
 
             input_parameters = {}
             for input_step_parameter in self.input_step_parameters:
-                label = input_step_parameter.workflow_step.label
-                if not label:
```

Comment:
You could use something like `f"{input_step_parameter.workflow_step}: Unnamed parameter"`. That's going to be unique enough I guess. In a followup I would suggest making the label field required in the editor.

Reply:
Ok, that sounds good.
Can I still make the label field required within the same PR, or should it be a new one that targets 24.1 or dev?

Reply:
That depends, I don't think this is straightforward as you have to check for uniqueness and nudge the user to fill in the label. I would prefer separate PRs.